### PR TITLE
Don't output an unused PrepareListFromTLV function.

### PR DESCRIPTION
**Pull request by bzbarsky-apple**
Only output it if it's needed.

#### Problem
We are outputting a function that is not used in some cases, and that fails to compile.

#### Change overview
Output the function conditionally.

#### Testing
@holbrookt says this fixes the compilation issues he was seeing.

### CODE REVIEW

**Review by holbrookt — APPROVED** [review]

**Review by woody-apple — APPROVED** [review]

**Review by andy31415 — CHANGES_REQUESTED** [review]
Requesting changes: adding complexity to zap files is undesirable if it can be avoided.

In this case I believe it can: the function can be moved into a header/cpp file. I believe there  is no significant gain of having it inside the cpp file as a static  function.

**Review by selissia — APPROVED** [review]

File: src/app/zap-templates/templates/app/CHIPClientCallbacks-src.zapt [hunk 1]
```diff
@@ -330,6 +330,7 @@ bool emberAfDiscoverCommandsReceivedResponseCallback(ClusterId clusterId, uint16
     return true;
 }
 
+{{#if (chip_has_client_cluster_with_server_list_attribute)}}
```
[the remark this answers is not in the document]
Comment (holbrookt) [reply src/app/zap-templates/templates/app/CHIPClientCallbacks-src.zapt:333]: That sounds good to me. @bzbarsky-apple are you tagging me just to get my thoughts on this suggestion?

**Comment by holbrookt** [comment]
@andy31415 
> In this case I believe it can: the function can be moved into a header/cpp file. I believe there is no significant gain of having it inside the cpp file as a static function.

I'm fine with this. Do you have a suggestion for an existing file or directory? Maybe `src/app/` ?

**Comment by holbrookt** [comment]
Filed https://github.com/project-chip/connectedhomeip/pull/9346 for the alternate fix suggested by @andy31415 . Suggest closing this PR once that one is approved

**Comment by andy31415** [comment]
#9346 was merged. Closing this PR based on comment from @holbrookt .

@bzbarsky-apple if this is wrong please re-open and comment on what the real status should be.

**Comment by bzbarsky-apple** [comment]
Yep, closing this is correct.  Thanks!

**Pull request closed without merging**